### PR TITLE
MCLOUD-5637: Update all error codes in ece-tools

### PR DESCRIPTION
**Pull request by oshmyheliuk**
## Purpose of this pull request

This pull request (PR) will add a list of possible error codes in the deployment process

## Affected DevDocs pages

- https://devdocs.magento.com/cloud/trouble/error-codes.html

## Magento source code

https://github.com/magento/ece-tools/pull/726

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: src/cloud/trouble/error-codes.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
+---
+group: cloud-guide
+title: Ece-tools error codes
+---
+
+This page helps you to detect the problem in case of failed deployment process.
+
+
```
Comment (NadiyaS) [inline src/cloud/trouble/error-codes.md:8]: are you sure that 2 new lines should be added?

File: src/cloud/trouble/error-codes.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+List of possible errors:
+
+### Build stage
+
+| Step | Description | Error code |
+| --- | --- | --- |
+| general | File `./app/etc/env.php` is not writable | [2](#ENV_PHP_IS_NOT_WRITABLE) |
+| | Configuration isn't defined in schema.yaml | [3](#CONFIG_NOT_DEFINED) |
+| | Failed to parse .magento.env.yaml | 4 |
```
Comment (NadiyaS) [inline src/cloud/trouble/error-codes.md:17]: won't it be suggestion to check that values in this files are in correct format?

Comment (NadiyaS) [reply src/cloud/trouble/error-codes.md:17]: I see that you added suggestion, but need to link it here ##### CONFIG_PARSE_FAILED

File: src/cloud/trouble/error-codes.md (hunk 1 above)
Comment (meker12) [inline src/cloud/trouble/error-codes.md:8]:
```suggestion
```

File: src/cloud/trouble/error-codes.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+| | Composer package was not found | 25 |
+
+### Deploy stage
+
+| Step | Description | Error code |
+| --- | --- | --- |
+| pre-deploy: cache | Wrong cache configuration (missed port or host) | 101 |
+| general | File `./app/etc/env.php` is not writable | 102 |
+| | Configuration isn't defined in schema.yaml | [103](#CONFIG_NOT_DEFINED) |
+| | Failed to parse .magento.env.yaml | 104 |
```
Comment (meker12) [inline src/cloud/trouble/error-codes.md:47]:
```suggestion
| | Failed to parse `.magento.env.yaml` file | 104 |
```

File: src/cloud/trouble/error-codes.md [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+| | Unable to read .magento.env.yaml | 105 |
```
Comment (meker12) [inline src/cloud/trouble/error-codes.md:48]: Change to all descriptions that mention the filename  --the `filename` file 

```suggestion
| | Unable to read the `.magento.env.yaml` file | 105 |
```

File: src/cloud/trouble/error-codes.md [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+| | Wrong search configuration | 111 |
+| | Wrong resource configuration | 112 |
+| validate-config:elasticsuite-integrity | ElasticSuite is installed without available ElasticSearch service | 113 |
+| | ElasticSuite is installed but another search engine is used | 114 |
+| general | Database query execution failed | 115 |
+| install-update: setup | Command `/bin/magento setup:install` failed | 116 |
+| install-update: config-import | Command `app:config:import` failed | 117 |
+| general | Required utility wasn't found (timeout, bash) | 118 |
+| install-update: deploy-static-content	 | Command `/bin/magento setup:static-content:deploy` failed | 119 |
+| compress-static-content | Compression of static content failed | 120 |
```
Comment (meker12) [inline src/cloud/trouble/error-codes.md:63]:
```suggestion
| compress-static-content | Failed to compress static content | 120 |
```

File: src/cloud/trouble/error-codes.md [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+| deploy-static-content:generate | Cannot update deployed version | 121 |
+| clean-static-content | Failed to clean static content files | 122 |
+| install-update: split-db | Command `/bin/magento setup:db-schema:split` failed | 123 |
+| clean-view-preprocessed | Failed to clean var/view_preprocessed | 124 |
```
Comment (meker12) [inline src/cloud/trouble/error-codes.md:67]:
```suggestion
| clean-view-preprocessed | Failed to clean the `var/view_preprocessed` folder | 124 |
```

File: src/cloud/trouble/error-codes.md [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+| install-update: reset-password | Failed to update `/var/credentials_email.txt` file | 125 |
+| install-update: update | Command `/bin/magento setup:upgrade` failed | 126 |
+| clean-cache | Command `/bin/magento cache:flush` failed | 127 |
+| disable-maintenance-mode | Command `/bin/magento maintenance:disable` failed | 128 | 
+| install-update: reset-password | Enable to read reset password template | 129 |
```
Comment (meker12) [inline src/cloud/trouble/error-codes.md:72]: This one is a little confusing --  What is the error?  That the reset password template can't be read?

File: src/cloud/trouble/error-codes.md [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+
+### Post-deploy stage
+
+| Step | Description | Error code |
+| --- | --- | --- |  
+| is-deploy-failed | If deploy stage failed (flag `deploy_is_failed` exists) | 201 |
```
Comment (meker12) [inline src/cloud/trouble/error-codes.md:78]:
```suggestion
| is-deploy-failed | Deploy stage failed If deploy stage failed | 201 |
```
or maybe :   The `deploy_is_failed` flag indicates failure during the deploy stage

File: src/cloud/trouble/error-codes.md [hunk 9]
```diff
@@ -0,0 +1,194 @@
...
+| general |  File `./app/etc/env.php` is not writable | 202 |
```
Comment (meker12) [inline src/cloud/trouble/error-codes.md:79]:
```suggestion
| general |  The `./app/etc/env.php` file is not writable | 202 |
```

File: src/cloud/trouble/error-codes.md [hunk 10]
```diff
@@ -0,0 +1,194 @@
...
+| | Configuration isn't defined in schema.yaml | 203 |
+| | Failed to parse .magento.env.yaml | 204 |
+| | Unable to read .magento.env.yaml | 205 |
+| | Unable to read .schema.yaml | 206 |
+| warm-up | Failed to warm-up some pages | 207 |
+| time-to-firs-byte | Failed to test time to first byte | 208 |
```
Comment (meker12) [inline src/cloud/trouble/error-codes.md:85]:
```suggestion
| time-to-firs-byte | Failed to test time to first byte (TTFB) | 208 |
```

File: src/cloud/trouble/error-codes.md [hunk 11]
```diff
@@ -0,0 +1,194 @@
...
+| clean-cache | Command `/bin/magento cache:flush` failed | 227 |
+
+## Errors suggestion
+
+##### ENV_PHP_IS_NOT_WRITABLE
+Deployment script can't make required changes in the `/app/etc/env.php` file. Check your filesystem permissions.
```
Comment (meker12) [inline src/cloud/trouble/error-codes.md:91]:
```suggestion
Deployment script can't make required changes to the `/app/etc/env.php` file. Check your filesystem permissions.
```

File: src/cloud/trouble/error-codes.md [hunk 12]
```diff
@@ -0,0 +1,194 @@
...
+| validate-config | Wrong database configuration | 109 |
+| | Wrong session configuration | 110 |
+| | Wrong search configuration | 111 |
+| | Wrong resource configuration | 112 |
+| validate-config:elasticsuite-integrity | ElasticSuite is installed without available ElasticSearch service | 113 |
+| | ElasticSuite is installed but another search engine is used | 114 |
+| general | Database query execution failed | 115 |
+| install-update: setup | Command `/bin/magento setup:install` failed | 116 |
+| install-update: config-import | Command `app:config:import` failed | 117 |
+| general | Required utility wasn't found (timeout, bash) | 118 |
```
Comment (meker12) [inline src/cloud/trouble/error-codes.md:61]: Error 117 should be included under the general section at line 58.

**Comment by meker12** [comment]
Changes merged into PR #7298.

**Pull request closed without merging**